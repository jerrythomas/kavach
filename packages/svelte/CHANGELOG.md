# @kavach/svelte

## 1.0.0-next.6

### Patch Changes

- Added store for auth status and used alerts to display status on demo page.
- Updated dependencies
  - @kavach/cookie@1.0.0-next.6
  - @kavach/core@1.0.0-next.6
  - @kavach/deflector@1.0.0-next.6
  - @kavach/logger@1.0.0-next.6

## 1.0.0-next.5

### Patch Changes

- chore: release 1.0.0-next.4
- Split core package into logger and deflector
- 1a35ad0: Resolved linter errors.
- Releasing the next version.
- Updated dependencies
- Updated dependencies
- Updated dependencies [1a35ad0]
- Updated dependencies
  - @kavach/cookie@1.0.0-next.5
  - @kavach/core@1.0.0-next.5
  - @kavach/deflector@1.0.0-next.5
  - @kavach/logger@1.0.0-next.5
