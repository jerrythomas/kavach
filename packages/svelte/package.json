{
	"name": "@kavach/svelte",
	"version": "1.0.0-next.2",
	"description": "Drop in security for web apps.",
	"author": "Jerry Thomas <me@jerrythomas.name>",
	"license": "MIT",
	"main": "index.js",
	"svelte": "src/index.js",
	"module": "src/index.js",
	"types": "dist/index.d.ts",
	"type": "module",
	"repository": {
		"type": "git",
		"url": "https://github.com/jerrythomas/kavach",
		"directory": "packages/kavach"
	},
	"prettier": "@jerrythomas/prettier-config",
	"publishConfig": {
		"access": "public"
	},
	"scripts": {
		"prepublishOnly": "tsc",
		"lint": "prettier --check --plugin-search-dir=. . && eslint .",
		"format": "prettier --write --plugin-search-dir=. .",
		"test:ci": "vitest run",
		"test:ui": "vitest --ui",
		"test": "vitest",
		"coverage": "vitest run --coverage",
		"upgrade": "pnpm upgrade"
	},
	"devDependencies": {
		"@jerrythomas/prettier-config": "^1.0.0",
		"@sveltejs/vite-plugin-svelte": "^2.0.2",
		"@testing-library/svelte": "^3.2.2",
		"@vitest/ui": "~0.12.10",
		"c8": "^7.12.0",
		"eslint-config-svelte": "workspace:*",
		"js-yaml": "^4.1.0",
		"jsdom": "^19.0.0",
		"shared-config": "workspace:^1.0.0",
		"svelte": "^3.55.0",
		"typescript": "^4.9.4",
		"vite": "^4.0.3",
		"vitest": "~0.19.1"
	},
	"exports": {
		"./package.json": "./package.json",
		".": {
			"types": "./types/index.d.ts",
			"import": "./src/index.js"
		}
	},
	"dependencies": {
		"@kavach/cookie": "workspace:1.0.0-next.2",
		"@kavach/core": "workspace:*",
		"@svelte-spice/form": "1.0.0-next.3",
		"@svelte-spice/input": "1.0.0-next.3",
		"ramda": "^0.28.0"
	}
}
