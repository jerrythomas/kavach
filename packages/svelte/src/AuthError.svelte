<script>
	export let isError = false
	export let message
	export let name
	export let status

	$: status = status ?? error_code
</script>

<error class="flex flex-col">
	<span>
		{#if status}
			{status}:
		{/if}
		{#if name}
			{name}:
		{/if}
		{message}
	</span>
	{#if message}
		<span>{message.replaceAll('+', ' ')}</span>
	{/if}
</error>
