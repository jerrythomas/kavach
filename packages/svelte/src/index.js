export * from './kavach'
export * from './request'
export * from './endpoints'

export { default as AuthButton } from './AuthButton.svelte'
export { default as AuthProvider } from './AuthProvider.svelte'
export { default as AuthGroup } from './AuthGroup.svelte'
export { default as AuthError } from './AuthError.svelte'
export { default as AuthResponse } from './AuthResponse.svelte'
export { default as AuthHandler } from './AuthHandler.svelte'
export { default as Logout } from './Logout.svelte'
