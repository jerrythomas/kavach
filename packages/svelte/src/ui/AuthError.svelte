<script>
	export let message
	export let name = null
	export let status = null
</script>

<error class="flex flex-col">
	<span>
		{#if status}
			{status}:
		{/if}
		{#if name}
			{name}:
		{/if}
		{message}
	</span>
</error>
