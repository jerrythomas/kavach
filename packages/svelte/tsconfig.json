{
  "extends": "shared-config/jsconfig.json",
  "include": ["src/**/*.js", "src/**/*.svelte"],
  "compilerOptions": {
    "outDir": "dist"
  }
}
