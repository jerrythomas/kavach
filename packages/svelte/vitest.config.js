import { defineConfig } from 'vite'
import { svelte } from '@sveltejs/vite-plugin-svelte'
import { testConfig } from 'shared-config'

testConfig.coverage.provider = 'v8'
export default defineConfig({
	plugins: [svelte({ hot: !process.env.VITEST })],
	test: testConfig
})
