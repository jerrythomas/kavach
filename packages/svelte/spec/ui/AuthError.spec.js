import { describe, expect, it, beforeEach } from 'vitest'
import { cleanup, render } from '@testing-library/svelte'
import { tick } from 'svelte'
import AuthError from '../../src/ui/AuthError.svelte'

describe('AuthError.svelte', () => {
	beforeEach(() => {
		cleanup()
	})

	it('should render with changes', async () => {
		const { container, component } = render(AuthError, {
			props: { message: 'An error occurred' }
		})
		expect(container).toMatchSnapshot()

		component.$$set({ message: 'Something else', name: 'AuthApi' })
		await tick()
		expect(container).toMatchSnapshot()

		component.$$set({ message: 'Another error', status: '400' })
		await tick()
		expect(container).toMatchSnapshot()
	})

	it('should render error with name', () => {
		const { container } = render(AuthError, {
			props: { message: 'An error', name: 'AuthApi' }
		})
		expect(container).toMatchSnapshot()
	})

	it('should render error with status', () => {
		const { container } = render(AuthError, {
			props: { message: 'An error', status: '400' }
		})
		expect(container).toMatchSnapshot()
	})

	it('should render error with status & name', () => {
		const { container } = render(AuthError, {
			props: { message: 'An error', status: '400', name: 'AuthApi' }
		})
		expect(container).toMatchSnapshot()
	})
})
