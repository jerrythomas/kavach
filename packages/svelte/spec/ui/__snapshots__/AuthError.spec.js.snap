// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthError.svelte > should render error with name 1`] = `
<div>
  <error
    class="flex flex-col"
  >
    <span>
       
      AuthApi
      :
       
      An error
    </span>
  </error>
</div>
`;

exports[`AuthError.svelte > should render error with status & name 1`] = `
<div>
  <error
    class="flex flex-col"
  >
    <span>
      400
      :
       
      AuthApi
      :
       
      An error
    </span>
  </error>
</div>
`;

exports[`AuthError.svelte > should render error with status 1`] = `
<div>
  <error
    class="flex flex-col"
  >
    <span>
      400
      :
       
       
      An error
    </span>
  </error>
</div>
`;

exports[`AuthError.svelte > should render with changes 1`] = `
<div>
  <error
    class="flex flex-col"
  >
    <span>
       
       
      An error occurred
    </span>
  </error>
</div>
`;

exports[`AuthError.svelte > should render with changes 2`] = `
<div>
  <error
    class="flex flex-col"
  >
    <span>
       
      AuthApi
      :
       
      Something else
    </span>
  </error>
</div>
`;

exports[`AuthError.svelte > should render with changes 3`] = `
<div>
  <error
    class="flex flex-col"
  >
    <span>
      400
      :
       
      AuthApi
      :
       
      Another error
    </span>
  </error>
</div>
`;
